src/providers/SpeechProvider/SpeechProvider.actions.js

```diff
@@ -124,15 +124,40 @@ export function changeRate(rate) {
 export function getVoices() {
   return dispatch => {
     dispatch(requestVoices());
-
     return tts.getVoices().then(pvoices => {
       let voices = [];
       try {
-        voices = pvoices.map(({ voiceURI, lang, name }) => ({
-          voiceURI,
-          lang,
-          name
-        }));
+        voices = pvoices.map(
+          ({
+            voiceURI,
+            lang,
+            name,
+            Locale,
+            ShortName,
+            DisplayName,
+            Gender
+          }) => {
+            let voice = {};
+            if (lang) {
+              voice.lang = lang;
+            } else if (Locale) {
+              voice.lang = Locale;
+            }
+            if (voiceURI) {
+              voice.voiceURI = voiceURI;
+              voice.voiceSource = 'local';
+            } else if (ShortName) {
+              voice.voiceURI = ShortName;
+              voice.voiceSource = 'cloud';
+            }
+            if (name) {
+              voice.name = name;
+            } else if (DisplayName) {
+              voice.name = `${DisplayName}  (${voice.lang}) - ${Gender}`;
```

Comment:
Very minor detail but there are two spaces before the opening parenthesis instead of one.